Tests/StarknetTests/Utils/DevnetClient.swift

```diff
@@ -0,0 +1,235 @@
+    var private_key: Felt
+    var public_key: Felt
+    var address: Felt
+    var salt: Felt
+}
+
+enum DevnetClientError: Error {
+    case invalidTestPlatform
+}
+
+// Due to DevnetClient being albe to run only on a macos, this
+// factory method will throw, when ran on any other platform.
+func makeDevnetClient() throws -> DevnetClientProtocol {
+#if os(macOS)
+    return DevnetClient()
+#endif
+    
+    throw DevnetClientError.invalidTestPlatform
+}
+
+#if os(macOS)
+
+class DevnetClient: DevnetClientProtocol {
+    private var host = "0.0.0.0"
+    private var port = 5050
+    private var seed = 1053545547
+    // Paths still arent working properly, for now I'm using absolute paths
+    private var accountDirectory = "/Users/jakub/test"
+    private var baseUrl: String
+   
+    private var isDevnetRunning = false
+    private var devnetProcess: Process!
+    
+    var gatewayUrl: String
+    var feederGatewayUrl: String
+    var rpcUrl: String
+    
+    struct TransactionResult {
+        var address: Felt
+        var hash: Felt
+    }
+    
+    struct DeployAccountResult{
+        var details: AccountDetails
+        var txHash: Felt
+    }
+    
+    
+    
+    init() {
+        baseUrl = "http://\(host):\(port)"
+        gatewayUrl = "\(baseUrl)/gateway"
+        feederGatewayUrl = "\(baseUrl)/feeder_gateway"
+        rpcUrl = "\(baseUrl)/rpc"
+    }
+    
+    public func start(){
+        
+        let arguments = "--host \(host) --port \(port) --seed \(seed)"
+        
+        // This kills any zombie devnet processes left over from previous
+        // test runs, if any.
+        let task = Process()
+        task.arguments = [
+            "-c",
+            "pkill -f starknet-devnet",
+            arguments]
+        
+        task.launchPath = "/bin/zsh"
+        task.launch()
+        task.waitUntilExit()
+        
+        // For some reason PATH used for executing shell commands in swift differs from
+        // PATH in the system. Currently full path to the program is needed
+        let command = "/Users/jakub/.asdf/shims/starknet-devnet"
+        //let command = "starknet-devnet"
+        
+        devnetProcess = Process()
+        let pipe = Pipe()
+            
+        devnetProcess.standardOutput = pipe
+        devnetProcess.standardError = pipe
+        devnetProcess.arguments = [
+            "-l",
+            "-c",
+            command,
+            arguments]
+        
+        devnetProcess.launchPath = "/bin/zsh"
+        devnetProcess.standardInput = nil
+        devnetProcess.launch()
+        
+        isDevnetRunning = true
+    }
+    
+    public func close(){
+        if(!isDevnetRunning){
+            return
+        }
+        
+        devnetProcess.terminate()
+        
+        // Wait for the process to be terminated
+        devnetProcess.waitUntilExit()
+        isDevnetRunning = false
+    }
+    
+    // needs finishing
+    public func prefundAccount(accountAddress: Felt) {
```

Comment:
Let's make it async (and throws as well) and use sessions' async await syntax, to wait for response